sdk/src/main/scala/io/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -267,11 +259,13 @@ class AccountForgeMessageBuilder(
               val withdrawalEpochNumber: Int = WithdrawalEpochUtils.getWithdrawalEpochInfo(mainchainBlockReferencesData.size, dummyView.getWithdrawalEpochInfo, params).epoch
 
               // get all previous payments for current ending epoch and append the one of the current block
-              val feePayments = dummyView.getFeePaymentsInfo(withdrawalEpochNumber, Some(currentBlockPayments))
+              val feePayments = dummyView.getFeePaymentsInfo(withdrawalEpochNumber, consensusEpochNumber, Some(currentBlockPayments))
```

Comment:
I am not sure, but couldn't we use the `consensusEpochNumber` of the `dummyView` instance instead of passing the one got from the blockContext? Should not they be the same?

Inside the AccountStateView class, something like:

    `val consensusEpochNumber: ConsensusEpochNumber = getConsensusEpochNumber.getOrElse(intToConsensusEpochNumber(0))`

Reply:
Ok I was wrong, the reason is that if we are on the edge of end-epoch the next block can result in having the next epoch value